### ISSUE TITLE
Picking stops working depending on the camera transform

### ISSUE THREAD

**Issue by tgecho** [aevyrie/bevy_mod_picking#117]
I'm working my way through [this bevy tutorial](https://caballerocoll.com/blog/bevy-chess-tutorial/) and have hit what seems to be a bug in bevy_mod_picking.

It didn't work initially with the tutorial's original camera config, which (with updates for bevy 0.6) looks like this:
```rs
PerspectiveCameraBundle{
    transform: Transform::from_matrix(Mat4::from_rotation_translation(
        Quat::from_xyzw(-0.3, -0.5, -0.3, 0.5).normalize(),
        Vec3::new(-7.0, 20.0, 4.0),
    )),
    ..Default::default()
}
```

After messing around with the minimal example from bevy_mod_picking, I finally found a version that works:
```
PerspectiveCameraBundle {
    transform: Transform::from_xyz(-6.0, 5., 3.5)
        .looking_at(Vec3::new(4.0, 0., 3.499), Vec3::Y),
    ..Default::default()
}
```

However, if I change the `z` of the `Vec3` to `3.5` (or greater), picking stops working. There's no debug cursor or logged events. Also, in the first camera version above, tweaking the rotation a bit can also make picking work again.

Thanks!

Full example here: https://gist.github.com/tgecho/a88a27825f2cb8b404cc0c4699e5bb5f
rust 1.60.0-nightly (17d29dcdc 2022-01-21)
bevy 0.6.0
bevy_mod_picking 0.5.2

**Comment by aevyrie** [comment]
Thanks for the repro, really helpful.

Looks like it breaks as soon as the `looking_at` and `translation` `z` values are the same. If they are both set to 3.5, does not work. If `looking_at` `z` is reduced to 3.49, it works. This seems to correspond with the resulting quaternion flipping signs:
```shell
    rotation: Quat(
        -0.36005276,
        -0.6083937,
        -0.3597256,
        0.60894704,
    ),
```
vs.
```shell
    rotation: Quat(
        0.35988933,
        0.6086705,
        0.35988933,
        -0.6086705,
    ),
```

The raycasting math was updated with bevy 0.6 due to the camera projection changing. I'm guessing that's where the bug is.

**Comment by aevyrie** [comment]
Fix made in `bevy_mod_raycast`, released to crates.io with version `0.3.7`.
Pushing an update to this library now, will be released as version `0.5.3`.

**Comment by tgecho** [comment]
Confirmed. Thanks! (for the fix and the awesome lib!)
